File: LeetCode/Easy/0705. Design HashSet.cpp
class MyHashSet {
public:
    vector<int> hs;   
    MyHashSet() {
        hs.reserve(1000001);
    }
    
    void add(int key) {
        hs[key] = 1;
    }
    
    void remove(int key) {
        hs[key] = 0;
    }
    
    bool contains(int key) {
        return (hs[key] == 1) ? true : false;
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */

File: LeetCode/Easy/0705. Design HashSet.py
class MyHashSet(object):

    def __init__(self):
        self.n = 1000
        self.ht = [[] for i in range(self.n)]
        

    def add(self, key):
        """
        :type key: int
        :rtype: None
        """
        if key not in self.ht[key%self.n]: self.ht[key%self.n].append(key)
        

    def remove(self, key):
        """
        :type key: int
        :rtype: None
        """
        if key in self.ht[key%self.n]: self.ht[key%self.n].remove(key)
        

    def contains(self, key):
        """
        :type key: int
        :rtype: bool
        """
        return (key in self.ht[key%self.n])
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)
